pkg/daemon/daemon.go

```diff
@@ -82,6 +85,10 @@ type Daemon struct {
 	kubeletHealthzEnabled  bool
 	kubeletHealthzEndpoint string
 
+	updateMu sync.Mutex // guards working (in Rust this would be enforced by the type system)
+	updateCond *sync.Cond // hold updateMu
```

Comment:
This doesn't seem used anywhere?

Reply:
Yeah my original thought was that in the future where we do live updates the signal handler could watch the cond and exit as soon as we weren't updating but...we're so far from live updates.